lib/error.js

```diff
@@ -0,0 +1,215 @@
+const axios = require('axios').default;
+
+/**
+ * Recursively searches an error for the stack trace.
+ *
+ * @function _findStackTrace
+ *
+ * @since 0.1.1
+ * @access private
+ * @memberOf console-to-slack
+ *
+ * @param {object} error  The error object.
+ *
+ * @returns {string} stack  The stack trace from the error.
+ */
+const _findStackTrace = (error) => {
+
+	let result = null;
+
+	if (error) {
+
+		if (error.stack) {
+			return error.stack;
+		}
+
+		for (const prop in error) {
+
+			if (error.hasOwnProperty(prop) &&
+				typeof error[prop] === 'object') {
+
+				result = _findStackTrace(error[prop]);
+
+				if (result) {
+					return result;
+				}
+
+			}
+
+		}
+
+	}
+
+	return result;
+
+};
+
+/**
+ * Parses an error to build appropriate response text for both
+ * NodeJS (express/koa) API response errors and normal JavaScript errors.
+ *
+ * @function _parseErrorObject
+ *
+ * @since 1.1.0
+ * @access private
+ * @memberOf console-to-slack
+ *
+ * @param {object} error  The error object.
+ *
+ * @returns {object} result          The result.
+ * @returns {string} result.pretext  The pretext message to send to Slack.
+ * @returns {string} result.text     The text message to send to Slack.
+ */
+const _parseErrorObject = (error) => {
+
+	let pretext = '', text = '';
+
+	// Check if the error is a NodeJS (express/koa) API response error
+	if (error.statusCode) {
+
+		pretext += `*StatusCode*: ${error.statusCode}\n`;
+
+		if (error.errorMessage) {
+			pretext += `*ErrorMessage*: ${error.errorMessage}\n`;
+		}
+
+		if (error.userMessage) {
+			pretext += `*UserMessage*: ${error.userMessage}\n`;
+		}
+
+		// Attempt to parse the error data array
+		if (error.errors) {
+			text += `${JSON.stringify(error.errors)}\n`;
```

Comment:
Can we add a `Errors:` prefix here? And maybe an additional new line character to separate it from the stack trace?